src/crimson/os/seastore/journal/segmented_journal.cc

```diff
@@ -96,8 +96,11 @@ SegmentedJournal::prep_replay_segments(
     }
   });
 
-  auto journal_tail = segments.rbegin()->second.journal_tail;
-  segment_provider.update_journal_tail_committed(journal_tail);
+  if (segments.rbegin()->second.journal_tail != JOURNAL_SEQ_NULL) {
+    segment_provider.update_journal_tail_committed(
+      segments.rbegin()->second.journal_tail);
+  }
```

Comment:
This change seems to be a consequence of there being non-journal segments in segments.  Perhaps it would be simpler to filter those out first?

Reply:
Or is this duplicated by the logic in SegmentCleaner::mount()?

Reply:
> Or is this duplicated by the logic in SegmentCleaner::mount()?

Yeah, this is duplicated, will remove it.